documentation/asciidoc/computers/compute-module/cm-emmc-flashing.adoc

```diff
@@ -1,144 +1,155 @@
 
-Please ensure you are not writing to any USB devices whilst the installer is running.
+==== on Linux
 
-* Download and run the https://github.com/raspberrypi/usbboot/raw/master/win32/rpiboot_setup.exe[Windows installer] to install the drivers and boot tool
-* Plug your host PC USB into the USB slave port, making sure you have setup the board as described above
-* Apply power to the board; Windows should now find the hardware and install the driver
-* Once the driver installation is complete, run the `RPiBoot.exe` tool that was previously installed
-* After a few seconds, the Compute Module eMMC will pop up under Windows as a disk (USB mass storage device)
+To set up software on a Linux host device:
 
-==== Building `rpiboot` on your host system.
+. Run the following command to install `rpiboot`:
++
+[source,console]
+----
+$ sudo apt install rpiboot
+----
 
-Instructions for building and running the latest release of `rpiboot` are documented in the https://github.com/raspberrypi/usbboot/blob/master/Readme.md#building[USB boot readme] on GitHub.
+. Connect the IO Board to power.
+
+. Then, run `rpiboot`:
++
+[source,console]
+----
+$ sudo rpiboot
+----
 
-==== Writing to the eMMC (Windows)
+. After a few seconds, the Compute Module should appear as a mass storage device. Check the `/dev/` directory, likely `/dev/sda` or `/dev/sdb`, for the device. Alternatively, run `lsblk` and search for a device with a storage capacity that matches the capacity of your Compute Module.
 
-After `rpiboot` completes, a new USB mass storage drive will appear in Windows. We recommend using https://www.raspberrypi.com/software/[Raspberry Pi Imager] to write images to the drive.
+TIP: Alternatively, you can https://github.com/raspberrypi/usbboot[build `rpiboot` from source].
 
-Make sure J4 (USB SLAVE BOOT ENABLE)/J2 (nRPI_BOOT) is set to the disabled position and/or nothing is plugged into the USB slave port. Power-cycling the IO board should now result in the Compute Module booting from eMMC.
+==== on macOS
 
-==== Writing to the eMMC (Linux)
+To set up software on a macOS host device:
 
-After `rpiboot` completes, you will see a new device appear; this is commonly `/dev/sda` on a Raspberry Pi but it could be in another location such as `/dev/sdb`, so check in `/dev/` or run `lsblk` before running `rpiboot` so you can see what changes.
+. First, https://github.com/raspberrypi/usbboot?tab=readme-ov-file#macos[build `rpiboot` from source].
 
-You now need to write a raw OS image (such as https://www.raspberrypi.com/software/operating-systems/#raspberry-pi-os-32-bit[Raspberry Pi OS]) to the device. Note the following command may take some time to complete, depending on the size of the image (change `/dev/sdX` to the appropriate device):
+. Connect the IO Board to power.
 
-[,bash]
+. Then, run the `rpiboot` executable with the following command:
++
+[source,console]
 ----
-sudo dd if=raw_os_image_of_your_choice.img of=/dev/sdX bs=4MiB
+$ sudo ./rpiboot
 ----
```

Comment:
The old version of the instructions suggested running `rpiboot` first and then powering up he IO board, but in the new instructions you seem to have switched that round? (for every OS)

Reply:
`rpiboot` requires the board to be powered to recognise the board as a mass storage device, AFAIK. See https://www.jeffgeerling.com/blog/2020/flashing-raspberry-pi-compute-module-on-macos-usbboot for third-party proof. Technically they could power the board _before_ they install the `usbboot` utility and dependencies, but I just wanted to reinforce that the board must be powered before you run `rpiboot`.